core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -820,6 +820,22 @@ static RelDataType deriveTypeSplit(SqlOperatorBinding operatorBinding,
           .withOperandTypeInference(InferTypes.RETURN_TYPE)
           .withKind(SqlKind.CONCAT2);
 
+  private static final SqlReturnTypeInference ARRAY_RETURN_TYPE =
```

Comment:
these days we use methods rather than lambda constants. see e.g. `transformConvert`.